commons/all/src/main/java/org/infinispan/commons/util/StringPropertyReplacer.java

```diff
@@ -63,6 +63,8 @@
     * <p/>
     * The property ${/} is replaced with System.getProperty("file.separator")
     * value and the property ${:} is replaced with System.getProperty("path.separator").
+    * <p/>
+    * Environment variables are referenced by the <code>env.</code> prefix, for example ${env.PATH}
```

Comment:
It's maybe for a follow up PR, but other than your CLI note, I don't think we describe property substitution for our various configurations anywhere in the docs. It would be good to document that 1) It's possible and 2) the various property types that are described in this Javadoc. \\cc @oraNod

Reply:
Indeed. In this PR I've fixed the lack of a test

Reply:
Thanks for cc'ing me, Ryan. agreed that docs could be expanded here with some examples. @tristantarrant would you prefer if I open a docs jira and do that async?

Reply:
Async is better for me as my PR now depends on this one :slightly_smiling_face:

Reply:
Perfect, I don't want to hold up the code. ISPN-11973